### PR TITLE
THREESCALE-10133: Add a cucumber for Current Utilization section

### PR DESCRIPTION
**Pull request by jlledom**
This comes from https://github.com/3scale/porta/pull/3558, I had to rename the branch because it's making a mess in Jira.

### CODE REVIEW

File: features/step_definitions/stats/application_steps.rb [hunk 1]
```diff
@@ -7,29 +7,16 @@
...
-
-  within(selector_for('the secondary nav')) do
-    click_on 'Analytics', match: :one
-  end
-
-  assert_text "Traffic statistics for #{developer_app_name}"
-end
-
-Given(/^buyer "(.*?)" made (\d+) service transactions (\d+) hours ago:$/) do |developer_name, transactions_number, hours, table|
+Given(/^the buyer made (\d+) service transactions (\d+) hours ago:$/) do |transactions_number, hours, table|
```
Comment (josemigallas) [inline features/step_definitions/stats/application_steps.rb:10]: Just a suggestion
```suggestion
Given "the buyer made {int} service transactions {int} hours ago:" do |transactions_number, hours, table|
```

**Review by josemigallas — APPROVED** [review]

File: features/provider/stats/application.feature [hunk 2]
```diff
@@ -4,16 +4,41 @@ Feature: Application stats
...
+
+  @javascript
+  Scenario: Developer's Current Utilization section has data
+    Given the provider has a free application plan "Limited"
+    And an usage limit on plan "Limited" for metric "hits" with period hour and value 100
+    And the buyer changed to application plan "Limited"
+    And the buyer makes 2 service transactions with:
+      | Metric   | Value |
+      | hits     |    20 |
+    And the backend will respond to a utilization request for the application with:
```
Comment (mayorova) [inline features/provider/stats/application.feature:30]: I guess I'd prefer "responds" rather than future tense.

Comment (jlledom) [reply features/provider/stats/application.feature:30]: Done

File: features/step_definitions/stats/application_steps.rb (hunk 1 above)
Comment (jlledom) [reply features/step_definitions/stats/application_steps.rb]: Done

File: features/provider/stats/application.feature [hunk 3]
```diff
@@ -4,16 +4,41 @@ Feature: Application stats
...
+
+  @javascript
+  Scenario: Developer's Current Utilization section has data
+    Given the provider has a free application plan "Limited"
+    And an usage limit on plan "Limited" for metric "hits" with period hour and value 100
+    And the buyer changed to application plan "Limited"
+    And the buyer makes 2 service transactions with:
+      | Metric   | Value |
+      | hits     |    20 |
+    And the backend responds to a utilization request for the application with:
```
Comment (mayorova) [inline features/provider/stats/application.feature:30]: IMO, we should ideally abstract the exact response from Backend, and hide it within the steps (cucumbers are more "business rules", so I guess we shouldn't get too technical). So maybe in the scenario make it a table:
```
      | period   | metric_name | max_value | current_value
      | minute   |    hits     | 100       | 40
```
and in the step transform it to JSON.

But I think it's probably fine :)

File: features/provider/stats/application.feature [hunk 4]
```diff
@@ -4,16 +4,41 @@ Feature: Application stats
...
+        "utilization": [{
+            "period": "minute",
+            "metric_name": "hits",
+            "max_value": 100,
+            "current_value": 40
+        }]
+      }
+      """
+    When they go to the provider application page
+    Then the Current Utilization panel should contain the following data:
```
Comment (mayorova) [inline features/provider/stats/application.feature:42]: @josemigallas always says we shouldn't use "should" :grimacing:

Comment (jlledom) [reply features/provider/stats/application.feature]: Done!

File: features/provider/stats/application.feature (hunk 3 above)
Comment (jlledom) [reply features/provider/stats/application.feature:30]: Done!

**Review by mayorova — APPROVED** [review]
Great job @jlledom you rock! :rock:

File: features/step_definitions/transaction_steps.rb [hunk 5]
```diff
@@ -22,3 +22,15 @@
...
 end
+
+Given("{buyer} made {int} service transactions {int} hours ago:") do |buyer, count, hours, table|
+  travel_to(hours.hours.ago)
+  access_user_sessions
+  step %(buyer "#{buyer.name}" makes #{count} service transactions with:), table
+end
+
+Given "the backend responds to a utilization request for the application with:" do |table|
+  json = { status: 'found', utilization: table.hashes }.to_json
```
[the remark this answers is not in the document]
Comment (jlledom) [reply features/step_definitions/transaction_steps.rb:33]: 😎

**Review by josemigallas — APPROVED** [review]

**Pull request merged**